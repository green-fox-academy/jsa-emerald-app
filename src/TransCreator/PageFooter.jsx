import React from 'react';
import moment from 'moment';
import {
  Text,
  Button,
  Footer,
  FooterTab,
  Toast,
} from 'native-base';
import { Platform } from 'react-native';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import styles from './Style';
import { addNewTransaction } from '../Stats/actionCreator';

const pressHandler = (amount, date, view, dispatch) => {
  if (!amount) {
    Toast.show({
      position: 'top',
      textStyle: { color: '#F4511E' },
      text: 'Please enter the amount',
      buttonText: 'Okay',
    });
    return;
  }

  dispatch(addNewTransaction(view, date, amount));
};

const PageFooter = ({ transAmount, transType, transDate }) => {
  const dispatch = useDispatch();
  return (
    <Footer style={Platform.OS === 'ios' ? styles.footerIOS : styles.footerAndroid}>
      <FooterTab style={styles.footerAndroid}>
        <Button />
        <Button
          bordered
          style={styles.confirmButton}
          onPress={() => pressHandler(transAmount, transDate, transType, dispatch)}
        >
          <Text style={styles.confirmButtonText}>Add</Text>
        </Button>
        <Button />
      </FooterTab>
    </Footer>
  );
};


PageFooter.propTypes = {
  transAmount: PropTypes.string,
  transType: PropTypes.string,
  transDate: PropTypes.number,
};

PageFooter.defaultProps = {
  transAmount: null,
  transType: 'Expense',
  transDate: moment().unix(),
};

export default PageFooter;
