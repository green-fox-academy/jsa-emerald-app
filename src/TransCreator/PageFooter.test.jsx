import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import PageFooter from './PageFooter';

describe('<PageFooter/>', () => {
  it('render', () => {
    const tree = renderer.create(<PageFooter createHandler={jest.fn} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Click event trigger', () => {
    const btnClick = jest.fn(() => {});
    const wrapper = shallow(<PageFooter createHandler={btnClick} />);
    wrapper.find('#btn-pageFooter-confirm').simulate('press');
    expect(btnClick).toHaveBeenCalled();
  });
});
