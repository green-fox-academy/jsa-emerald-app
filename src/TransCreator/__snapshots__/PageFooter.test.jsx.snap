// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageFooter/> render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F8F8F8",
        "borderColor": "#cbcbcb",
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "center",
        "left": 0,
        "paddingBottom": 0,
        "right": 0,
      },
      Object {
        "backgroundColor": "transparent",
        "borderTopWidth": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": undefined,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#5C6BC0",
          "borderLeftWidth": 1,
          "borderRadius": 5,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "borderWidth": 1,
          "elevation": null,
          "flex": 1,
          "flexDirection": null,
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 30,
          "marginRight": 30,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#007aff",
              "fontFamily": "System",
              "fontSize": 14,
              "lineHeight": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "color": "#5C6BC0",
              "fontSize": 15,
            },
          ]
        }
        uppercase={false}
      >
        Add
      </Text>
    </View>
  </View>
</View>
`;
